excel/tasks.py

```diff
@@ -1,14 +1,23 @@
-import sys
+from importlib.machinery import SourceFileLoader
 from pathlib import Path
 
-# Add the devutils even if the poetry env isn't setup (to do a 'inv devinstall').
+# Since the `robocorp-devutils` dependency is available in the local Poetry isolated
+#  environment only, we won't have it available with Invoke commands running outside
+#  such environment.
+tasks_path = Path(__file__).absolute()
+invoke_utils_path = (
+    tasks_path.parent.parent /
+    "devutils" / "src" / "devutils" / "invoke_utils.py"
+)
+assert invoke_utils_path.is_file(), f"{invoke_utils_path} does not exist"
+
+# But we still need to import the package in order to expose the common Invoke tasks.
 try:
-    import devutils
+    from devutils import invoke_utils
 except ImportError:
-    devutils_src = Path(__file__).absolute().parent.parent / "devutils" / "src"
-    assert devutils_src.exists(), f"{devutils_src} does not exist."
-    sys.path.append(str(devutils_src))
-
-from devutils.invoke_utils import build_common_tasks
+    invoke_utils = SourceFileLoader(
```

Comment:
I'm not sure this is an improvement... it works because the `invoke_utils` doesn't import anything outside of that file at this point, but it'll break when that happens.

Reply:
Correct, I missed that!

Just that changing the `sys.path` at runtime (and such change has to be maintained in a dozen of different _tasks.py_ files) didn't look like a better idea than this newer suggestion I added. (and improved post-review)

----

I added a few more fixes to strengthen the solution and make it even cleaner so that:
- Any tasks.py package doesn't even need to know about the files/modules we write, it is sufficient now to refer to: `"devutils" / "src" / "devutils" / "__init__.py"`
- Any "publicly" exposed function at package init level is directly usable by any importer: `devutils.build_common_tasks()`
- The _invoke_utils.py_ file is the only one in charge now for managing `sys.path` with a path relative to itself, so we can do imports like: `from devutils.rcc import build_subprocess_kwargs`

What do you think now, isn't [this](https://github.com/robocorp/robocorp/blob/chore/dx-improvements/excel/tasks.py#L15-L21) more cleaner, robust and easier to maintain now in all those _tasks.py_ files we have in every package?

Reply:
Reverted as it was proved that `devutils` isn't expected to be a cohesive Python package and my change didn't add value.